compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -45,31 +45,29 @@ static LogicalResult applyBufferizeToAllocation(RewriterBase &rewriter,
 /// elementwise op's input is the output of the matmul op and has already been
 /// promoted, there is no need to promote such operand again.
 static SmallVector<Value> getInputOutputOperands(linalg::LinalgOp &linalgOp) {
-  if (isMatmulProducerOfElementwise(linalgOp)) {
-    SmallVector<Value> operands;
-    for (Value operand : linalgOp->getOperands()) {
-      if (isMatmulInDefChain(operand)) continue;
-      operands.push_back(operand);
-    }
-    return operands;
+  SmallVector<Value> operands;
+  for (Value operand : linalgOp->getOperands()) {
+    if (!isa<RankedTensorType>(operand.getType())) continue;
+    if (isMatmulProducerOfElementwise(linalgOp) && isMatmulInDefChain(operand))
```

Comment:
Would it suffice if we just use `isMatmulInDefChain` instead ?

Trying to understand case which would align with the following but we do not want to `continue` :-
> since one of the elementwise op's input is the output of the matmul op and has already been promoted, there is no need to promote such operand again.

Reply:
Good point. It doesn't really need `isMatmulProducerOfElementwise(linalgOp)` check, but I'd add a check to make sure the linalgOp is an elementwise op and then the matmul op is in its def chain. So I change it to 

`if (isElementwise(linalgOp) && isMatmulInDefChain(operand))`